.github/workflows/torch-wheel.yml

```diff
@@ -0,0 +1,53 @@
+name: Build PyTorch Wheel
+on:
+  workflow_dispatch:
+    inputs:
+        pytorch-version:
+            description: 'Version of pytorch to build (e.g. 2.1.2)'
+            required: true
+            default: '2.1.2'
+            type: string
+        push-to-registry:
+            description: 'Push image to registry'
+            required: true
+            type: boolean
+            default: true
+
+jobs:
+  build-pytorch-wheel:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Print Inputs
+      run: echo "${{ toJSON(github.event.inputs) }}"
+  
+    - id: created
+      run: echo "created=$(date --utc +%Y-%m-%dT%H:%M:%SZ)" >> $GITHUB_OUTPUT
+
+    - uses: actions/checkout@v4
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@f95db51fddba0c2d1ec667646a06c2ce06100226 # v3.0.0
+
+    - name: Log in to Docker Hub
+      uses: docker/login-action@343f7c4344506bcbf9b4de18042ae17996df046d # v3.0.0
+      if: github.ref == 'refs/heads/main'
+      with:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+    - uses: docker/build-push-action@4a13e500e55cf31b7a5d59a38ab2040ab0f42f56 # v5.1.0
+      with:
+        context: .
+        file: ./torch-wheel.Dockerfile
+        push: ${{ inputs.push-to-registry }}
+        build-args: |
+          TORCH_BUILD_VERSION=${{ inputs.pytorch-version }}
+        # TODO add the latest tag, maybe somehow via docker/metadata-action
+        tags: |
+            ${{ secrets.DOCKERHUB_USERNAME }}/torch-wheel-arm64:${{ inputs.pytorch-version }}
+        # https://docs.docker.com/build/ci/github-actions/cache/
+        cache-from: type=gha,scope=torch-wheel
+        cache-to: type=gha,scope=torch-wheel,mode=max
+        platforms: linux/arm64
```

Comment:
It might be a bit cleaner if we were able to build the amd64 image as well. That way the arm64 wheel would not be pulled for amd64 build.

It would be possible to have separate build stages for the architectures. Something like this might work in Dockerfile to build for ARM, but not for AMD:

```
ARG BASE_IMAGE
FROM scratch AS amd64

FROM ${BASE_IMAGE} AS arm64
...

FROM ${TARGETPLATFORM#linux/}

```

Reply:
I think we can use [$TARGETARCH](https://docs.docker.com/reference/dockerfile/#automatic-platform-args-in-the-global-scope), perhaps more readable than the variable expansion magic.

I updated the wheel Dockerfile to support both platforms. Just a dummy for x86 and for arm we copy the wheel from opukka/torch-wheels image. I hope this is acceptable for now, until we have the appropriate runners available.

Reply:
True. I'm too used to the TARGETPLATFORM and the shell expansion syntax.